samples/CommunityToolkit.Maui.Sample/ViewModels/Converters/ByteArrayToImageSourceConverterViewModel.cs

```diff
@@ -0,0 +1,69 @@
+﻿using CommunityToolkit.Mvvm.Input;
+
+namespace CommunityToolkit.Maui.Sample.ViewModels.Converters;
+
+public sealed class ByteArrayToImageSourceConverterViewModel : BaseViewModel, IDisposable
+{
+	readonly HttpClient client;
+
+	bool isDownloadingImage;
+	byte[]? dotNetBotImageByteArray;
+	string labelText = "Tap the Download Image Button to download an Image as a byte[]";
+
+	public ByteArrayToImageSourceConverterViewModel(HttpClient httpClient)
+	{
+		client = httpClient;
+		DownloadDotNetBotImageCommand = new AsyncRelayCommand(ExecuteDownloadDotNetBotImageCommand, () => !IsDownloadingImage && dotNetBotImageByteArray is null, false);
+	}
+
+	public AsyncRelayCommand DownloadDotNetBotImageCommand { get; }
+
+	public bool IsDownloadingImage
+	{
+		get => isDownloadingImage;
+		set
+		{
+			SetProperty(ref isDownloadingImage, value);
+			MainThread.BeginInvokeOnMainThread(DownloadDotNetBotImageCommand.NotifyCanExecuteChanged);
+		}
+	}
+
+	public byte[]? DotNetBotImageByteArray
+	{
+		get => dotNetBotImageByteArray;
+		set => SetProperty(ref dotNetBotImageByteArray, value);
+	}
+
+	public string LabelText
+	{
+		get => labelText;
+		set => SetProperty(ref labelText, value);
+	}
+
+	public void Dispose()
+	{
+		DotNetBotImageByteArray = null;
+	}
+
+	async Task ExecuteDownloadDotNetBotImageCommand()
+	{
+		IsDownloadingImage = true;
+
+		// Ensure Activity Indicator appears on screen for a minumum of 1.5 seconds when the uses taps the Download Button
+		var minimumDownloadTimeTask = Task.Delay(TimeSpan.FromSeconds(1.5));
+
+		try
+		{
+			DotNetBotImageByteArray = await client.GetByteArrayAsync("https://user-images.githubusercontent.com/13558917/137551073-ac8958bf-83e3-4ae3-8623-4db6dce49d02.png").ConfigureAwait(false);
```

Comment:
Can we use image from Embedded Resources? To avoid using HttpClient and extra dependency and potential issue if there is no the Internet connection

Reply:
Ah, I like injecting `HttpClient! 

I think it's a good example that we can share with devs on how to add `HttpClient` to `IServiceCollection` via `builder.Services.AddHttpClient()`, then how to inject it into a ViewModel.

And the `Microsoft.Extensions.Http` dependency is only added to the sample app, not to the `CommunityToolkit.Maui` or `CommunityToolkit.Maui.Core` libraries.

Good call on the lack of internet connectivity, though! I'll add a `catch` block to ensure the app doesn't crash when there's no internet connectivity 👍